specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json

```diff
@@ -0,0 +1,355 @@
+    "AutoDismissAlertsRulesList": {
+      "type": "object",
+      "description": "Auto dismiss rules list for subscription.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AutoDismissAlertsRule"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "AutoDismissAlertsRule": {
+      "type": "object",
+      "description": "Describes the auto dismiss rule",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AutoDismissAlertsRuleProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "AutoDismissAlertsRuleProperties": {
+      "type": "object",
+      "description": "describes AutoDismissAlertsRule properties",
+      "properties": {
+        "alertType": {
+          "type": "string",
+          "description": "Type of the auto dismissed alert. For all alert types, use '*'"
+        },
+        "lastModifiedUTC": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The last time this rule was modified"
+        },
+        "expirationDateUTC": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Expiration date of the rule"
+        },
+        "reason": {
+          "type": "string",
+          "description": "The reason for dismissing the alert"
+        },
+        "state": {
+          "type": "string",
+          "description": "Possible states of the rule",
+          "x-ms-enum": {
+            "name": "State"
+          },
+          "enum": [
+            "Enabled",
+            "Disabled",
+            "Expired"
+          ]
+        },
+        "comment": {
+          "type": "string",
+          "description": "Any comment regarding the rule"
+        },
+        "autoDismissAlertsScope": {
+          "type": "object",
+          "description": "The auto dismiss conditions",
+          "$ref": "#/definitions/AutoDismissAlertsScope"
+        },
+        "ruleImpact": {
+          "type": "object",
+          "description": "The impact of the rule. Use for testing the rule and will be returned only when the test flag is set",
+          "readOnly": true,
+          "$ref": "#/definitions/RuleImpact"
+        }
+      },
+      "required": [
+        "alertType",
+        "reason",
+        "state"
+      ]
+    },
+    "AutoDismissAlertsScope": {
+      "type": "object",
+      "properties": {
+        "allOf": {
+          "type": "array",
+          "description": "All the conditions inside need to be true in order to auto dismiss the alert",
+          "items": {}
```

Comment:
Hi,
I didn't find a way to model the items, so i'm keeping it as {}.
If you have a way, i'll be glad to hear. 
My items should be a mixed types array (which is not supported at V2).
Discriminator won't help me because I don't have enum for one field , to distinguish.
Those are my types:
{
"field": string,
"in": array [strings]
}
{
"field": string,
"contains": string
}

I've tried to model "Field" as base class for items, but it didn't worked
    "Field": {
      "type": "object",
      "x-abstract": true,
      "properties": {
        "field": {
          "type": "string"
         }
      },
      "required": [
        "field"
      ]
    },
    "FieldIn": {
      "type": "object",
      "x-abstract": false,
      "x-inherits": "#/definitions/Field",
      "properties": {
        "field": {
          "type": "string"
        },
        "in": {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      },
      "required": [
        "field",
        "in"
      ]
    },
    "FieldContains": {
      "type": "object",
      "x-abstract": false,
      "x-inherits": "#/definitions/Field",
      "properties": {
        "field": {
          "type": "string"
        },
        "contains": {
          "type": "string"
        }
      },
      "required": [
        "field",
        "contains"
      ]
    }

Do you have a suggestion how to model the items? Instead of keeping them as {}?

Reply:
@KrisBash Hi Kris, do you have an idea for this?